helix-core/src/main/java/org/apache/helix/task/TaskUtil.java

```diff
@@ -1045,4 +1048,54 @@ private static void setNextJobPurgeTime(String workflow, long currentTime, long
       rebalanceScheduler.scheduleRebalance(manager, workflow, nextPurgeTime);
     }
   }
+
+  /**
+   * Get target thread pool size from InstanceConfig first; if that fails, get it from
+   * ClusterConfig; if that fails, fall back to the default value.
+   * @param configAccessor - accessor used for the configs
+   * @param clusterName - the cluster name for InstanceConfig and ClusterConfig
+   * @param instanceName - the instance name for InstanceConfig
+   * @return target thread pool size
+   */
+  public static int getTargetThreadPoolSize(ConfigAccessor configAccessor, String clusterName,
```

Comment:
You have 2 very similar code blocks, especially the warning messages. Is it possible to make it concise? I understand the warning messages might look the same in that case, but it won't hurt IMO. Since we just need to know that the configuration is missing.
Duplicate code is a bigger problem.

Reply:
This is the first thing I tried with this PR, and my conclusion was there is no way to not have duplicate lines here. The main reason is due to the "try catch" + "optional return" structure, making it very difficult to factor out. Also, since the getters have different signatures, that's an additional difficulty to factor it out. If you have a good way to make it not duplicate, I'd love to try it.